public/app/features/datasources/state/reducers.ts

```diff
@@ -17,13 +17,14 @@ export const initialState: DataSourcesState = {
   searchQuery: '',
   dataSourcesCount: 0,
   dataSourceTypeSearchQuery: '',
-  hasFetched: false,
   isLoadingDataSources: false,
+  isLoadingDataSourcePlugins: false,
   dataSourceMeta: {} as DataSourcePluginMeta,
   isSortAscending: true,
 };
 
 export const dataSourceLoaded = createAction<DataSourceSettings>('dataSources/dataSourceLoaded');
+export const dataSourcesLoad = createAction<void>('dataSources/dataSourcesLoad');
```

Comment:
Naming is a bit confusing to me (maybe it's only me). Something like `dataSourcesLoadingStarted` would be a bit clearer but I see you're just using the pattern used in other actions so you don't need to change it (unless you also feel it's confusing and you want to change other names as well).

Reply:
yeah, i was also kinda bothered by it, but didn't want to create inconsistencies within that file and i'd rather not mix refactors and actual changes, but happy to do it if you feel strongly about it.